airbyte-webapp/src/pages/BuilderPage/components/YamlEditor.tsx

```diff
@@ -0,0 +1,67 @@
+import Editor, { Monaco } from "@monaco-editor/react";
+import { useState } from "react";
+import { useDebounce, useLocalStorage } from "react-use";
+
+import { DownloadYamlButton } from "./DownloadYamlButton";
+import styles from "./YamlEditor.module.scss";
+import { template } from "./YamlTemplate";
+
+export const YamlEditor: React.FC = () => {
+  const [locallyStoredEditorValue, setLocallyStoredEditorValue] = useLocalStorage<string>(
+    "connectorBuilderEditorContent",
+    template
+  );
+  const [editorValue, setEditorValue] = useState(locallyStoredEditorValue ?? "");
+  useDebounce(() => setLocallyStoredEditorValue(editorValue), 500, [editorValue]);
+
+  const handleEditorChange = (value: string | undefined) => {
+    setEditorValue(value ?? "");
+  };
+
+  const setEditorTheme = (monaco: Monaco) => {
+    monaco.editor.defineTheme("airbyte", {
+      base: "vs-dark",
+      inherit: true,
+      rules: [
+        { token: "string", foreground: styles.tokenString },
+        { token: "type", foreground: styles.tokenType },
+        { token: "number", foreground: styles.tokenNumber },
+        { token: "delimiter", foreground: styles.tokenDelimiter },
+        { token: "keyword", foreground: styles.tokenKeyword },
+      ],
+      colors: {
+        "editor.background": "#0d0d2d",
```

Comment:
This blue is darker than our current `_colors.scss` darkest blue, which is why I hardcoded it here.

I got this value from the [design](https://www.figma.com/file/F1ljvrJerLoVbNvtejBWid/LOW-CODE-CONNECTOR-BUILDER?node-id=1141%3A42453#274512475) -- should we add this color to `_colors.scss`?

Reply:
I actually figured out a way to get the gradient from the design to show in the editor (couldn't do this before because monaco editor.background only accepts a hex value)

I did it by setting the monaco editor background to transparent (using an 8-character hex value with the last two "alpha" digits set to 00), and applied a gradient to the background of the editor. See this commit for the changes: https://github.com/airbytehq/airbyte/pull/17482/commits/b15769a28469a209fa58faa8f267fe038708c1a4

@Upmitt I tried to match the gradient as close as possible to the design, this is a screenshot of what it looks like in my branch:
<img width="1245" alt="Screen Shot 2022-10-11 at 3 00 18 PM" src="https://user-images.githubusercontent.com/22731524/195206219-1a68ed52-2325-4693-bf5b-e438ec14ab6e.png">

But I think Tim's question still applies. Should we update our color palette to include these gradient background colors? Or should they just be directly used in this one case, like I've done in this branch?

Reply:
Hi @timroes  & @lmossman ! I can't see where the #0d0d2d color appears in the designs on Figma, but maybe it's an error. Did you find it using a color picker, or by inspecting Figma? I only use our referenced colors, also for gradient, and if there is other colors, it's an error I did. The values I use for the dark blues are #1A194D and #0A0A23 - But please let me know if I introduced a new one by mistake :) - I think this gradient #1A194D > #0A0A23 may be used in other screen, it is the gradient I used in the website V2.0

Reply:
@Upmitt I tried to find the #0d0d2d color in the prototype again but I'm not finding it, so I'm not sure where I got that value in the first place 🤷 

But I will fix this to be a gradient of #1A194D > #0A0A23

#1A194D currently exists in our color scheme as `dark-blue-900`, but I'm not seeing #0A0A23 anywhere in our color scheme. Should I go ahead and add that to our list of colors [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/scss/_colors.scss)? What should the name of that color be? `dark-blue-1000`?